src/Umbraco.Infrastructure/Packaging/AutomaticPackgeMigrationPlan.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Xml.Linq;
+using Umbraco.Cms.Core.Packaging;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Infrastructure.Migrations;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Infrastructure.Packaging
+{
+    /// <summary>
+    /// Used to automatically indicate that a package has an embedded package data manifest that needs to be installed
+    /// </summary>
+    public abstract class AutomaticPackgeMigrationPlan : PackageMigrationPlan
+    {
+        protected AutomaticPackgeMigrationPlan(string name)
+            : base(name)
+        {
+        }
+
+        protected sealed override void DefinePlan()
+        {
+            // calculate the final state based on the hash value of the embedded resource
+            Type planType = GetType();
+            PackageDataManifest = PackageMigrationResource.GetEmbeddedPackageDataManifest(planType);
```

Comment:
I wonder if this should just happen in the getter of the `PackageDataManifest` and then save the result in a backing field? Seems safer to me if we in the future wanna use  `PackageDataManifest` before we call call `DefinePlan`

Reply:
yep makes sense